pytorch_lightning/trainer/connectors/precision_connector.py

```diff
@@ -79,3 +93,16 @@ def connect(self, model):
             self.trainer.optimizers = optimizers
 
         return model
+
+    def _check_sharded_plugin(self, plugins):
+        if plugins and self._sharded_in_plugins(plugins):
+            if not FAIRSCALE_AVAILABLE:
```

Comment:
Is it the right place for `MisconfigurationException` ? I would have excepted this to be raised while creating `DDPShardedPlugin` already.